packages/marketplace/contracts/TransferManager.sol

```diff
@@ -202,15 +202,48 @@ abstract contract TransferManager is Initializable, ITransferManager {
     /// @param paymentSide DealSide of the fee-side order
     /// @param nftSide DealSide of the nft-side order
     function _doTransfersWithFeesAndRoyalties(DealSide memory paymentSide, DealSide memory nftSide) internal {
-        uint256 fees;
         uint256 remainder = paymentSide.asset.value;
+        (, address nftSideRecipient) = _getRecipients(paymentSide, nftSide);
+
+        (uint256 fees, bool shouldTransferRoyalties) = _calculateFeesAndRoyalties(nftSide);
+        bool isBundle = nftSide.asset.assetType.assetClass == LibAsset.AssetClass.BUNDLE;
+
+        if (isBundle) {
+            if (!_isTSBSeller(nftSide.account)) {
+                remainder = _doTransfersWithFeesAndRoyaltiesForBundle(paymentSide, nftSide, nftSideRecipient);
+            } else {
+                // No royalties but primary fee should be paid on the total value of the bundle
+                remainder = _transferProtocolFees(remainder, paymentSide, fees);
+            }
+        } else if (shouldTransferRoyalties) {
+            remainder = _transferRoyalties(remainder, paymentSide, nftSide);
+        }
+        if (!isBundle) {
+            remainder = _transferProtocolFees(remainder, paymentSide, fees);
```

Comment:
Small thing, after we merge other PRs to this lets add a comment above this line about why we do this

`// Transfer protocol fees for non bundle asset, bundles pay per item protocol fees`